sdk/lib/core/uri.dart

```diff
@@ -24,24 +24,52 @@ const int _LOWER_CASE_Z = 0x7A;
 
 const String _hexDigits = "0123456789ABCDEF";
 
-/// A parsed URI, such as a URL.
+/// A parsed URI, such as a URL. URI is Uniform Resource Identifier. URL is
+/// Uniform Resource Locator, a.k.a. a web address.
 ///
-/// To create a URI with specific components, use [new Uri]:
+/// Use this class to represent web addresses (`https://...`), local file
+/// locations (`C:\file` on Windows or `/file` on Unixes), or local folder
+/// locations (`C:\folder\` or `/folder/`).
+///
+/// You can create web URIs either by specifying the scheme (protocol) as a 
+/// parameter (`Uri(scheme: 'https', ...)`) or using the optional [Uri.https]
+/// or [Uri.http] constructor.
+///
+/// To create a URI, use [new Uri]:
 /// ```dart
-/// var httpsUri = Uri(
+/// // Use a final if the Uri contents never change
+/// final dartLibTour = Uri(
 ///     scheme: 'https',
 ///     host: 'dart.dev',
 ///     path: '/guides/libraries/library-tour',
 ///     fragment: 'numbers');
-/// print(httpsUri); // https://dart.dev/guides/libraries/library-tour#numbers
+/// print(dartLibTour); // https://dart.dev/guides/libraries/library-tour#numbers
+/// ```
 ///
-/// httpsUri = Uri(
+/// The optional parts of the URL at the end come in two flavors: they can
```

Comment:
Might be worth introducing the *format* of a *hierarchical URI* (which this class is generally representing),
before talking about the optional parts at the end.

```dart
/// A hierarchical URI has the following format:
/// ```
/// scheme://username:password@host:port/path?query#fragment
/// ```
/// where all parts are optional. 
```

If that precedes the "To create a URI", it explains the names of the parameters of `Uri`.